prime-router/src/main/kotlin/azure/ValidateFunction.kt

```diff
@@ -0,0 +1,328 @@
+                    ActionLog(InvalidReportMessage(e.message ?: "Invalid request."), type = ActionLogLevel.error)
+                )
+                HttpStatus.BAD_REQUEST
+            }
+
+        val submission = DetailedSubmissionHistory(
+            1,
+            TaskAction.none,
+            OffsetDateTime.now(),
+            httpStatus.value(),
+            mutableListOf<DetailedReport>(),
+            actionHistory.actionLogs.map {
+                DetailedActionLog(
+                    it.scope,
+                    it.reportId,
+                    it.index,
+                    it.trackingId,
+                    it.type,
+                    it.detail
+                )
+            }
+        )
+
+        // set status for validation response
+        submission.overallStatus = if (httpStatus == HttpStatus.BAD_REQUEST)
+            DetailedSubmissionHistory.Status.ERROR
+        else
+            DetailedSubmissionHistory.Status.VALID
+
+        return request.createResponseBuilder(httpStatus)
+            .header(HttpHeaders.CONTENT_TYPE, "application/json")
+            .body(
+                JacksonMapperUtilities.allowUnknownsMapper
+                    .writeValueAsString(submission)
+            )
+            .build()
+    }
+
+    /**
+     * Extract client header from request headers or query string parameters
+     * @param request the http request message from the client
+     */
+    private fun extractClient(request: HttpRequestMessage<String?>): String {
+        // client can be in the header or in the url parameters:
+        return request.headers[CLIENT_PARAMETER]
+            ?: request.queryParameters.getOrDefault(CLIENT_PARAMETER, "")
+    }
+
+    /**
+     * Extract the optional payloadName (aka sender-supplied filename) from request headers or query string parameters
+     * @param request the http request message from the client
+     */
+    private fun extractPayloadName(request: HttpRequestMessage<String?>): String? {
+        // payloadName can be in the header or in the url parameters.  Return null if not found.
+        return request.headers[PAYLOAD_NAME_PARAMETER]
+            ?: request.queryParameters[PAYLOAD_NAME_PARAMETER]
+    }
+
+    private fun processingType(request: HttpRequestMessage<String?>, sender: Sender): Sender.ProcessingType {
+        val processingTypeString = request.queryParameters[PROCESSING_TYPE_PARAMETER]
+        return if (processingTypeString == null) {
+            sender.processingType
+        } else {
+            try {
+                Sender.ProcessingType.valueOfIgnoreCase(processingTypeString)
+            } catch (e: IllegalArgumentException) {
+                sender.processingType
+            }
+        }
+    }
+
+    /**
+     * Take the request and parse it into a [ValidatedRequest] object with some
+     * sensible default values
+     */
+    internal fun validateRequest(request: HttpRequestMessage<String?>): ValidatedRequest {
```

Comment:
This function is an identical duplicate with the one in ReportFunction. Perhaps full out into a shared base class?

Reply:
@clediggins-usds Took a stab at this, I called the new base class "RequestFunction" but I'm certainly open to suggestions there (and otherwhere. Otherwhere? Yes, otherwhere. I know that's not the queen's English but it outta be)